flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/table/converter/ArrayAttributeConverterProvider.java

```
@@ -0,0 +1,80 @@
+        } else if (enhancedType.equals(EnhancedType.of(BigDecimal[].class))) {
+            return (AttributeConverter<T>)
+                    getArrayAttributeConverter(BigDecimal.class, enhancedType);
+        } else if (enhancedType.equals(EnhancedType.of(Byte[].class))) {
+            return (AttributeConverter<T>) getArrayAttributeConverter(Byte.class, enhancedType);
+        } else if (enhancedType.equals(EnhancedType.of(Short[].class))) {
+            return (AttributeConverter<T>) getArrayAttributeConverter(Short.class, enhancedType);
+        } else if (enhancedType.equals(EnhancedType.of(Integer[].class))) {
+            return (AttributeConverter<T>) getArrayAttributeConverter(Integer.class, enhancedType);
+        } else if (enhancedType.equals(EnhancedType.of(Long[].class))) {
+            return (AttributeConverter<T>) getArrayAttributeConverter(Long.class, enhancedType);
+        } else if (enhancedType.equals(EnhancedType.of(Float[].class))) {
+            return (AttributeConverter<T>) getArrayAttributeConverter(Float.class, enhancedType);
+        } else if (enhancedType.equals(EnhancedType.of(LocalDate[].class))) {
+            return (AttributeConverter<T>)
+                    getArrayAttributeConverter(LocalDate.class, enhancedType);
+        } else if (enhancedType.equals(EnhancedType.of(LocalTime[].class))) {
+            return (AttributeConverter<T>)
+                    getArrayAttributeConverter(LocalTime.class, enhancedType);
+        } else if (enhancedType.equals(EnhancedType.of(LocalDateTime[].class))) {
+            return (AttributeConverter<T>)
+                    getArrayAttributeConverter(LocalDateTime.class, enhancedType);
+        } else if (enhancedType.equals(EnhancedType.of(Instant[].class))) {
+            return (AttributeConverter<T>) getArrayAttributeConverter(Instant.class, enhancedType);
+        }
+        return null;
```

What happens in this failure mode? Should we throw an exception here?

As discussed, you missed `Double`